specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/adaptiveNetworkHardenings.json

```diff
@@ -0,0 +1,377 @@
+                ],
+                "operationId": "AdaptiveNetworkHardenings_ListByExtendedResource",
+                "parameters": [
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+                    },
+                    {
+                        "name": "resourceNamespace",
+                        "in": "path",
+                        "description": "The Namespace of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceType",
+                        "in": "path",
+                        "description": "The type of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "resourceName",
+                        "in": "path",
+                        "description": "Name of the resource.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/AdaptiveNetworkHardeningsList"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Security/adaptiveNetworkHardenings/{adaptiveNetworkHardeningResourceName}": {
+            "get": {
+                "x-ms-examples": {
+                    "Get a single Adaptive Network Hardening resource": {
+                        "$ref": "./examples/AdaptiveNetworkHardenings/GetAdaptiveNetworkHardening_example.json"
+                    }
+                },
+                "tags": [
+                    "AdaptiveNetworkHardenings"
+                ],
+                "description": "Gets a single Adaptive Network Hardening resource",
+                "operationId": "AdaptiveNetworkHardening_Get",

```

Comment:
Should be `AdaptiveNetworkHardenings_Get` (missing 's').

Reply:
Why? it gets only a single resource, not a list

Reply:
For a given operation ID, AutoRest generates an operation group (e.g. an interface in C#) that's named to everything left of the '_' character that will contain all of the methods in the same group.  So if you have the following operation groups.
```
Foos_Get
Foos_CreateOrUpdate
Foos_Delete
```
You'd get the following codegen.
```C#
public interface Foo {
    Get();
    CreateOrUpdate();
    Delete();
}
```
The other problem here is that operation group name `AdaptiveNetworkHardening` collides with the type `AdaptiveNetworkHardening` defined on line 324; this will force the code generators to disambiguate the name by whatever means the generators have implemented (usually some generic prefix or suffix) which is not very optimal.  It's for this reason that the authoring guidelines state that for operation groups NOUN_VERB the noun be plural so the model type can be singular.